### PR TITLE
Improve unit test coverage for MSQ

### PR DESCRIPTION
**Pull request by LakshSingla**
### Description

This PR adds a new set of tests called `MSQFaultsTest` that aims to improve the testing coverage for MSQ by adding new unit tests for when the query should fault.

<hr>

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [x] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Review by adarshsanjeev — APPROVED** [review]
Thanks for the PR! 
Tests added look good to me. Should there also be tests for `TooManyInputFilesFault` etc here?

**Review by adarshsanjeev — APPROVED** [review]
New test case looks good as well. Thanks for making the changes!

**Comment by kfaraz** [comment]
@LakshSingla , it seems a real overkill to have 30k rows of data just for to get a specific exception in a unit test.
Can we simplify this somehow?

**Comment by LakshSingla** [comment]
@kfaraz MAX_PARTITIONS (25k) is a constant defined above which the query would fail. To get this many partitions, in my understanding we would require at least 25000 rows. I can think of the following options to prevent this static file here:
1. Create a file programmatically and then run the tests.
2. Create a method and override MAX_WORKERS (which doesn't seem right to me to do for the sake of testing).
3. Use some Guice magic to inject MAX_WORKERS (similar reasoning as above). 
Do you have any opinions on the above or an alternate approach that you think might be suitable? Thanks!

**Comment by kfaraz** [comment]
Yes, @LakshSingla , the first option seems most reasonable. You shouldn't have to generate the file either. You could maybe add a utility method in `MsqTestBase` which gives you a mock file with the required number of rows and columns?

File: extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java [hunk 1]
```diff
@@ -462,6 +464,27 @@ protected File getResourceAsTemporaryFile(final String resource) throws IOExcept
     return file;
   }
 
+  /**
+   * Helper method that populates a temporary file with {@code numRows} rows and {@code numColumns} columns where the
+   * first column is a string 'timestamp' while the rest are string columns with junk value
+   */
+  protected File generateTemporaryNdJsonFile(final int numRows, final int numColumns) throws IOException
```
Comment (cryptoe) [inline extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/MSQTestBase.java:471]: I think we can add a new class for this and the previous method. MSQTestFileUtils?

**Review by cryptoe — APPROVED** [review]
Thanks @LakshSingla !!

**Pull request merged**